python/ribasim_testmodels/ribasim_testmodels/invalid.py

```diff
@@ -195,3 +195,119 @@ def invalid_fractional_flow_model():
     )
 
     return model
+
+
+def invalid_control_states_model():
+    xy = np.array(
+        [
+            (0.0, 0.0),  # 1: Basin
+            (1.0, 0.0),  # 2: Pump
+            (2.0, 0.0),  # 3: LevelBoundary
+            (1.0, 1.0),  # 4: DiscreteControl
+        ]
+    )
+    node_xy = gpd.points_from_xy(x=xy[:, 0], y=xy[:, 1])
+
+    node_type = ["Basin", "Pump", "LevelBoundary", "DiscreteControl"]
+
+    # Make sure the feature id starts at 1: explicitly give an index.
+    node = ribasim.Node(
+        static=gpd.GeoDataFrame(
+            data={"type": node_type},
+            index=pd.Index(np.arange(len(xy)) + 1, name="fid"),
+            geometry=node_xy,
+            crs="EPSG:28992",
+        )
+    )
+
+    # Setup the edges:
+    from_id = np.array([1, 2, 4], dtype=np.int64)
+    to_id = np.array([2, 3, 2], dtype=np.int64)
+    lines = ribasim.utils.geometry_from_connectivity(node, from_id, to_id)
+    edge = ribasim.Edge(
+        static=gpd.GeoDataFrame(
+            data={
+                "from_node_id": from_id,
+                "to_node_id": to_id,
+                "edge_type": ["flow", "flow", "control"],
+            },
+            geometry=lines,
+            crs="EPSG:28992",
+        )
+    )
+
+    # Setup the basins:
+    profile = pd.DataFrame(
+        data={
+            "node_id": [1, 1],
+            "area": [0.01, 1.0],
+            "level": [0.0, 1.0],
+        }
+    )
+
+    static = pd.DataFrame(
+        data={
+            "node_id": [1],
+            "drainage": [0.0],
+            "potential_evaporation": [0.0],
+            "infiltration": [0.0],
+            "precipitation": [0.0],
+            "urban_runoff": [0.0],
+        }
+    )
+
+    # Setup pump:
+    pump = ribasim.Pump(
+        static=pd.DataFrame(
+            data={
+                "node_id": [2],
+                "flow_rate": [0.5 / 3600],
+            }
+        )
+    )
```

Comment:
Is the error here that this Pump has no state at all? Maybe nicer to set a state to "bar" here?

Reply:
And it would be helpful for the future to indicate (in a comment) where the error in the code is.